frontend/src/modules/dataset/dataset-browser.tsx

```diff
@@ -0,0 +1,389 @@
+                                            ...state.filter,
+                                            filteringTextDisplay: filteringText,
+                                            filteringText
+                                        }
+                                    };
+
+                                    // if there is a a prefix then the actual filter text should not include the prefix
+                                    if (filteringTextPrefix) {
+                                        payload.filter.filteringText = resourceForPath(filteringText); 
+                                    }
+
+                                    setState({
+                                        type: DatasetActionType.State,
+                                        payload
+                                    });
+
+                                    if (shouldFetch) {
+                                        // if the filter contains a prefix append that to the Location
+                                        const effectiveContext = {...state.datasetContext, location: [state.datasetContext.location, filteringTextPrefix].join('')};
+                                        fetchDatasetResources(effectiveContext);
+                                    }
+                                }
+                            }}
+                            onKeyDown={(event) => {
+                                if (event.detail.key.match(/[^\w!-.*'()/]/)) {
+                                    event.preventDefault();
+                                }
+                            }}
+                        />
+                    }
+                    isItemDisabled={(item) => {
+                        switch (displayMode) {
+                            case DatasetBrowserDisplayMode.Resource:
+                                return (item.type === 'prefix' && !isSelectingPrefix) || (item.type === 'object' && !isSelectingObject);
+                            case DatasetBrowserDisplayMode.Dataset:
+                                return !isSelectingPrefix;
+                            case DatasetBrowserDisplayMode.Scope:
+                                return true;
+                        }
+                    }}
+                    items={state.filter.filteredItems.slice((state.pagination.currentPageIndex - 1) * pageSize, (state.pagination.currentPageIndex) * pageSize)}
+                    loading={state.isLoading}
+                    loadingText={`Loading ${displayMode}s`}
+                    onSelectionChange={(event) => {
+                        setState({
+                            type: DatasetActionType.State,
+                            payload: {
+                                selectedItems: event.detail.selectedItems
+                            }
+                        });
+
+                        props.onSelectionChange?.(event);
+                    }}
+                    pagination={
```

Comment:
Why are we not using default paging?

Reply:
See my comment above about not being able to use the useCollection hook.

Reply:
I think we covered this in our dev sync about the limitations of the original.